diff --git a/node_modules/@sveltejs/kit/src/core/config/index.js b/node_modules/@sveltejs/kit/src/core/config/index.js
index ae35a70..d480e23 100644
--- a/node_modules/@sveltejs/kit/src/core/config/index.js
+++ b/node_modules/@sveltejs/kit/src/core/config/index.js
@@ -95,7 +95,7 @@ export async function load_config({ cwd = process.cwd() } = {}) {
  * @param {import('@sveltejs/kit').Config} config
  * @returns {import('types').ValidatedConfig}
  */
-function process_config(config, { cwd = process.cwd() } = {}) {
+export function process_config(config, { cwd = process.cwd() } = {}) {
 	const validated = validate_config(config);
 
 	validated.kit.outDir = path.resolve(cwd, validated.kit.outDir);
diff --git a/node_modules/@sveltejs/kit/src/exports/vite/index.js b/node_modules/@sveltejs/kit/src/exports/vite/index.js
index 5480800..5c81d67 100644
--- a/node_modules/@sveltejs/kit/src/exports/vite/index.js
+++ b/node_modules/@sveltejs/kit/src/exports/vite/index.js
@@ -87,7 +87,7 @@ const options_regex = /(export\s+const\s+(prerender|csr|ssr|trailingSlash))\s*=/
 const warned = new Set();
 
 /** @type {import('svelte/compiler').PreprocessorGroup} */
-const warning_preprocessor = {
+export const warning_preprocessor = {
 	script: ({ content, filename }) => {
 		if (!filename) return;
 
@@ -186,7 +186,7 @@ let build_metadata = undefined;
  * @param {{ svelte_config: import('types').ValidatedConfig }} options
  * @return {Promise<import('vite').Plugin[]>}
  */
-async function kit({ svelte_config }) {
+export async function kit({ svelte_config }) {
 	/** @type {import('vite')} */
 	const vite = await import_peer('vite');
 
@@ -1129,6 +1129,7 @@ async function kit({ svelte_config }) {
 					const bundle = /** @type {import('vite').Rollup.RollupOutput} */ (
 						await vite.build({
 							configFile: vite_config.configFile,
+							...(globalThis.__axiumNestedConfig ?? {}),
 							// CLI args
 							mode: vite_config_env.mode,
 							logLevel: vite_config.logLevel,
