--- a/src/exports/vite/index.js
+++ b/src/exports/vite/index.js
@@ -230,9 +230,9 @@
  *
  * @param {{ svelte_config: import('types').ValidatedConfig }} options
  * @return {Promise<Plugin[]>}
  */
-async function kit({ svelte_config }) {
+export async function kit({ svelte_config }) {
 	/** @type {typeof import('vite')} */
 	const vite = await import_peer('vite');
 
 	// @ts-ignore `vite.rolldownVersion` only exists in `vite 8`
@@ -1346,8 +1346,9 @@
 
 						const bundle = /** @type {import('vite').Rollup.RollupOutput} */ (
 							await vite.build({
 								configFile: vite_config.configFile,
+								...(globalThis.__axiumNestedConfig ?? {}),
 								// CLI args
 								mode: vite_config_env.mode,
 								logLevel: vite_config.logLevel,
 								clearScreen: vite_config.clearScreen,
