--- a/lib/_tsc.js	
+++ b/lib/_tsc.js	
@@ -7547,8 +7547,9 @@
   An_expanded_version_of_this_information_showing_all_possible_compiler_options: diag(6928, 3 /* Message */, "An_expanded_version_of_this_information_showing_all_possible_compiler_options_6928", "An expanded version of this information, showing all possible compiler options"),
   Compiles_the_current_project_with_additional_settings: diag(6929, 3 /* Message */, "Compiles_the_current_project_with_additional_settings_6929", "Compiles the current project, with additional settings."),
   true_for_ES2022_and_above_including_ESNext: diag(6930, 3 /* Message */, "true_for_ES2022_and_above_including_ESNext_6930", "`true` for ES2022 and above, including ESNext."),
   List_of_file_name_suffixes_to_search_when_resolving_a_module: diag(6931, 1 /* Error */, "List_of_file_name_suffixes_to_search_when_resolving_a_module_6931", "List of file name suffixes to search when resolving a module."),
+  Import_JSON_files_as_const_assertions: diag(6933, 3 /* Message */, "Import_JSON_files_as_const_assertions_6933", "Import JSON files as const assertions."),
   false_unless_checkJs_is_set: diag(6932, 3 /* Message */, "false_unless_checkJs_is_set_6932", "`false`, unless `checkJs` is set"),
   Variable_0_implicitly_has_an_1_type: diag(7005, 1 /* Error */, "Variable_0_implicitly_has_an_1_type_7005", "Variable '{0}' implicitly has an '{1}' type."),
   Parameter_0_implicitly_has_an_1_type: diag(7006, 1 /* Error */, "Parameter_0_implicitly_has_an_1_type_7006", "Parameter '{0}' implicitly has an '{1}' type."),
   Member_0_implicitly_has_an_1_type: diag(7008, 1 /* Error */, "Member_0_implicitly_has_an_1_type_7008", "Member '{0}' implicitly has an '{1}' type."),
@@ -37603,8 +37604,17 @@
     description: Diagnostics.Enable_importing_json_files,
     defaultValueDescription: false
   },
   {
+    name: "importJsonAsConst",
+    type: "boolean",
+    affectsSemanticDiagnostics: true,
+    affectsBuildInfo: true,
+    category: Diagnostics.Language_and_Environment,
+    description: Diagnostics.Import_JSON_files_as_const_assertions,
+    defaultValueDescription: false
+  },
+  {
     name: "allowArbitraryExtensions",
     type: "boolean",
     affectsProgramStructure: true,
     category: Diagnostics.Modules,
@@ -56664,8 +56674,11 @@
     if (isSourceFile(declaration) && isJsonSourceFile(declaration)) {
       if (!declaration.statements.length) {
         return emptyObjectType;
       }
+      if (compilerOptions.importJsonAsConst) {
+        return checkExpression(declaration.statements[0].expression);
+      }
       return getWidenedType(getWidenedLiteralType(checkExpression(declaration.statements[0].expression)));
     }
     if (isAccessor(declaration)) {
       return getTypeOfAccessors(symbol);
@@ -80718,8 +80731,11 @@
     return false;
   }
   function isConstContext(node) {
     const parent = node.parent;
+    if (compilerOptions.importJsonAsConst && parent.kind === 245 /* ExpressionStatement */ && isSourceFile(parent.parent) && isJsonSourceFile(parent.parent)) {
+      return true;
+    }
     return isAssertionExpression(parent) && isConstTypeReference(parent.type) || isJSDocTypeAssertion(parent) && isConstTypeReference(getJSDocTypeAssertionType(parent)) || isValidConstAssertionArgument(node) && isConstTypeVariable(getContextualType(node, 0 /* None */)) || (isParenthesizedExpression(parent) || isArrayLiteralExpression(parent) || isSpreadElement(parent)) && isConstContext(parent) || (isPropertyAssignment(parent) || isShorthandPropertyAssignment(parent) || isTemplateSpan(parent)) && isConstContext(parent.parent);
   }
   function checkExpressionForMutableLocation(node, checkMode, forceTuple) {
     const type = checkExpression(node, checkMode, forceTuple);
--- a/lib/typescript.d.ts	
+++ b/lib/typescript.d.ts	
@@ -7042,8 +7042,9 @@
         experimentalDecorators?: boolean;
         forceConsistentCasingInFileNames?: boolean;
         ignoreDeprecations?: string;
         importHelpers?: boolean;
+        importJsonAsConst?: boolean;
         /** @deprecated */
         importsNotUsedAsValues?: ImportsNotUsedAsValues;
         inlineSourceMap?: boolean;
         inlineSources?: boolean;
--- a/lib/typescript.js	
+++ b/lib/typescript.js	
@@ -10953,8 +10953,9 @@
   An_expanded_version_of_this_information_showing_all_possible_compiler_options: diag(6928, 3 /* Message */, "An_expanded_version_of_this_information_showing_all_possible_compiler_options_6928", "An expanded version of this information, showing all possible compiler options"),
   Compiles_the_current_project_with_additional_settings: diag(6929, 3 /* Message */, "Compiles_the_current_project_with_additional_settings_6929", "Compiles the current project, with additional settings."),
   true_for_ES2022_and_above_including_ESNext: diag(6930, 3 /* Message */, "true_for_ES2022_and_above_including_ESNext_6930", "`true` for ES2022 and above, including ESNext."),
   List_of_file_name_suffixes_to_search_when_resolving_a_module: diag(6931, 1 /* Error */, "List_of_file_name_suffixes_to_search_when_resolving_a_module_6931", "List of file name suffixes to search when resolving a module."),
+  Import_JSON_files_as_const_assertions: diag(6933, 3 /* Message */, "Import_JSON_files_as_const_assertions_6933", "Import JSON files as const assertions."),
   false_unless_checkJs_is_set: diag(6932, 3 /* Message */, "false_unless_checkJs_is_set_6932", "`false`, unless `checkJs` is set"),
   Variable_0_implicitly_has_an_1_type: diag(7005, 1 /* Error */, "Variable_0_implicitly_has_an_1_type_7005", "Variable '{0}' implicitly has an '{1}' type."),
   Parameter_0_implicitly_has_an_1_type: diag(7006, 1 /* Error */, "Parameter_0_implicitly_has_an_1_type_7006", "Parameter '{0}' implicitly has an '{1}' type."),
   Member_0_implicitly_has_an_1_type: diag(7008, 1 /* Error */, "Member_0_implicitly_has_an_1_type_7008", "Member '{0}' implicitly has an '{1}' type."),
@@ -41886,8 +41887,17 @@
     description: Diagnostics.Enable_importing_json_files,
     defaultValueDescription: false
   },
   {
+    name: "importJsonAsConst",
+    type: "boolean",
+    affectsSemanticDiagnostics: true,
+    affectsBuildInfo: true,
+    category: Diagnostics.Language_and_Environment,
+    description: Diagnostics.Import_JSON_files_as_const_assertions,
+    defaultValueDescription: false
+  },
+  {
     name: "allowArbitraryExtensions",
     type: "boolean",
     affectsProgramStructure: true,
     category: Diagnostics.Modules,
@@ -61299,8 +61309,11 @@
     if (isSourceFile(declaration) && isJsonSourceFile(declaration)) {
       if (!declaration.statements.length) {
         return emptyObjectType;
       }
+      if (compilerOptions.importJsonAsConst) {
+        return checkExpression(declaration.statements[0].expression);
+      }
       return getWidenedType(getWidenedLiteralType(checkExpression(declaration.statements[0].expression)));
     }
     if (isAccessor(declaration)) {
       return getTypeOfAccessors(symbol);
@@ -85353,8 +85366,11 @@
     return false;
   }
   function isConstContext(node) {
     const parent2 = node.parent;
+    if (compilerOptions.importJsonAsConst && parent2.kind === 245 /* ExpressionStatement */ && isSourceFile(parent2.parent) && isJsonSourceFile(parent2.parent)) {
+      return true;
+    }
     return isAssertionExpression(parent2) && isConstTypeReference(parent2.type) || isJSDocTypeAssertion(parent2) && isConstTypeReference(getJSDocTypeAssertionType(parent2)) || isValidConstAssertionArgument(node) && isConstTypeVariable(getContextualType2(node, 0 /* None */)) || (isParenthesizedExpression(parent2) || isArrayLiteralExpression(parent2) || isSpreadElement(parent2)) && isConstContext(parent2) || (isPropertyAssignment(parent2) || isShorthandPropertyAssignment(parent2) || isTemplateSpan(parent2)) && isConstContext(parent2.parent);
   }
   function checkExpressionForMutableLocation(node, checkMode, forceTuple) {
     const type = checkExpression(node, checkMode, forceTuple);
